### PR TITLE
Business logic layer implementation

### PR DESCRIPTION
**Pull request by Konstantsiy**

### CODE REVIEW

File: internal/repository/repository.go [hunk 1]
```diff
@@ -17,7 +17,13 @@ var (
 	ErrNoSuchRequest     = errors.New("request with the given id does not exists")
 )
 
-const uniqueViolationCode = "23505"
+const (
+	uniqueViolationCode = "23505"
+
+	RequestStatusProcessed = "processed"
+	RequestStatusFailed    = "failed"
+	RequestStatusDone      = "done"
```
Comment (papahmsolo) [inline internal/repository/repository.go:25]: What is the difference between `processed` and `done`?

Comment (Konstantsiy) [reply internal/repository/repository.go:25]: > What is the difference between `processed` and `done`?

This was written in the Technical Task :) 
Maybe this should show that the converter has already started processing and we could track the stage of readiness of the request.

Comment (papahmsolo) [reply internal/repository/repository.go]: Then let's call it `processing` or `in_progress`

**Review by papahmsolo — APPROVED** [review]
Let's merge it like this and get the other logic implemented, but we'll need to get back to this and address the issues of this approach

**Pull request merged**